src/components/ng2-map.component.ts

```diff
@@ -114,7 +114,7 @@ export class Ng2MapComponent implements OnChanges, OnDestroy, AfterViewInit {
       window['ng2MapRef'].forEach( ng2MapRef => {
         ng2MapRef.zone.run(function() { ng2MapRef.componentFn(); });
       });
-      window['ng2MapRef'] = [];
+      window['ng2MapRef'].splice(0, window['ng2MapRef'].length);
```

Comment:
I still didn't get it. Why do we need map object when we empty out `window.ng2MapRef`?